contracts/mock/MockERC20.sol

```diff
@@ -73,7 +73,40 @@ contract MockLUSD is MockERC20 {
 }
 
 contract MockLQTY is MockERC20 {
+    bool shouldFailOnApprove;
+    bool shouldFailOnTransfer;
+
     constructor(uint256 supply)
         MockERC20("Mock LQTY", "mockLQTY", 18, supply)
     {}
+
+    function setShouldFailOnApprove(bool _shouldFail) public {
```

Comment:
I'm not sure this is necessary. [Approve is always true](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L136).

Reply:
we have checks for this in the contract code, and I added this to test those checks. if that is not necessary, should we have those checks in the first place?

Reply:
i've [removed](https://github.com/lindy-labs/sc_solidity-contracts/pull/225/commits/f6bdd815baac5ff2f9355828898471602b940f46) those as we discussed in chat